test: test_invalid_command[py_3.14-invalidCommand] fails on Python 3.14.5 due to argparse quoting change

## Description

The `tests/test_cli.py::test_invalid_command[py_3.14-invalidCommand]` test started failing on Python 3.14.5 (released [May 10, 2026](https://www.python.org/downloads/release/python-3145/)) because CPython restored quoting of choices in argparse error messages:

> [gh-130750](https://github.com/python/cpython/issues/130750): Restore quoting of choices in [argparse](https://docs.python.org/3.14/library/argparse.html#module-argparse) error messages for improved clarity and consistency with documentation.
> -- [3.14.5 changelog](https://docs.python.org/3.14/whatsnew/changelog.html#python-3-14-5-final)

| Python | argparse "invalid choice" format |
| --- | --- |
| 3.10, 3.11 | `(choose from 'init', 'commit', ...)` (quoted) |
| 3.12 | mixed |
| 3.13, 3.14.0–3.14.4 | `(choose from init, commit, ...)` (unquoted) |
| 3.14.5+ | `(choose from 'init', 'commit', ...)` (quoted, restored) |

The checked-in fixture `tests/test_cli/test_invalid_command_py_3_14_invalidCommand_.txt` was generated against 3.14.4 (unquoted), so it no longer matches.

## Reproduction

```bash
uv run pytest tests/test_cli.py::test_invalid_command
```

Master's last successful CI run was on 2026-05-09 (commit `1eb8cde6`) when the GitHub-hosted runner still had Python 3.14.4; runs on or after 2026-05-11 fail with the diff below.

## Observed failure

```
FAILED tests/test_cli.py::test_invalid_command[py_3.14-invalidCommand] - AssertionError: FILES DIFFER
-cz: error: argument {...}: invalid choice: 'invalidCommand' (choose from init, commit, ...)
+cz: error: argument {...}: invalid choice: 'invalidCommand' (choose from 'init', 'commit', ...)
```

Full log: https://github.com/commitizen-tools/commitizen/actions/runs/25649120897/job/75283769681

## Scope

Only the **1** `test_invalid_command_py_3_14_invalidCommand_.txt` fixture is affected by gh-130750 (the companion `--invalid-arg` case uses the metavar, not the choice list). The other 60+ Python-version-keyed fixtures (`test_command_shows_description_when_use_help_option_py_3_*_*.txt`, `test_no_argv_py_3_*_.txt`) capture argparse **help text**, which is unaffected by this change.

## Suggested fix (short-term)

**Skip the affected `test_invalid_command` parametrization on Python ≥ 3.14.5** until upstream stabilizes:

```python
@pytest.mark.parametrize(
    "arg",
    [
        "--invalid-arg",
        pytest.param(
            "invalidCommand",
            marks=pytest.mark.skipif(
                (3, 14, 5) <= sys.version_info < (3, 15),
                reason=(
                    "argparse error format changed in Python 3.14.5 (gh-130750); "
                    "fixture matches 3.14.0-4 unquoted format"
                ),
            ),
        ),
    ],
)
```

Surfaced while working on #1846; this skip unblocks CI without changing PR scope. Implemented in #1991.

## Future enhancement (still open after #1991)

Argparse error output has churned across Python patch releases (3.10/3.11 quoted → 3.13/3.14.0–4 unquoted → 3.14.5 quoted). A more durable approach worth considering once the codebase is more stable:

**Normalize argparse error output before `file_regression.check()`** — e.g., strip surrounding quotes around `choose from` items, or replace the variable parts with placeholders:

```python
def _normalize_argparse_error(text: str) -> str:
    # Collapse "'init', 'commit'" and "init, commit" to a single canonical form
    text = re.sub(
        r"\(choose from [^)]+\)",
        "(choose from <CHOICES>)",
        text,
    )
    return text

# Usage:
file_regression.check(_normalize_argparse_error(err), extension=".txt")
```

That would let us delete the per-Python-version fixtures for `test_invalid_command[invalidCommand]` and stop chasing argparse formatting changes. **#1991 only implements the short-term skip; this enhancement is still open.**

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -1,4 +1,5 @@
 import os
+import re
 import subprocess
 import sys
 import types
@@ -17,6 +18,29 @@ from commitizen.exceptions import (
 )
 from tests.utils import UtilFixture
 
+ARGPARSE_CHOICES_PATTERN = re.compile(r"\(choose from (?P<choices>[^)]*)\)")
+ARGPARSE_QUOTED_CHOICE_PATTERN = re.compile(r"'([^']*)'")
+
+
+def normalize_argparse_choice_quotes(text: str) -> str:
+    def normalize_match(match: re.Match[str]) -> str:
+        choices = ARGPARSE_QUOTED_CHOICE_PATTERN.sub(r"\1", match.group("choices"))
+        return f"(choose from {choices})"
+
+    return ARGPARSE_CHOICES_PATTERN.sub(normalize_match, text)
+
+
+def test_normalize_argparse_choice_quotes():
+    text = (
+        "cz: error: argument {init,commit}: invalid choice: 'invalidCommand' "
+        "(choose from 'init', 'commit')"
+    )
+
+    assert normalize_argparse_choice_quotes(text) == (
+        "cz: error: argument {init,commit}: invalid choice: 'invalidCommand' "
+        "(choose from init, commit)"
+    )
+
 
 @pytest.mark.usefixtures("python_version", "consistent_terminal_output")
 def test_no_argv(util: UtilFixture, capsys, file_regression):
@@ -29,29 +53,7 @@ def test_no_argv(util: UtilFixture, capsys, file_regression):
 
 @pytest.mark.parametrize(
     "arg",
-    [
-        "--invalid-arg",
-        pytest.param(
-            "invalidCommand",
-            marks=pytest.mark.skipif(
-                (3, 14, 5) <= sys.version_info < (3, 15),
-                reason=(
-                    "Python 3.14.5 restored argparse choice quoting (CPython "
-                    "gh-130750); the checked-in fixture matches the 3.14.0-4 "
-                    "unquoted format. See #1990."
-                ),
-            ),
-        ),
-    ],
-)
-@pytest.mark.skipif(
-    sys.version_info[:2] == (3, 12) and sys.version_info < (3, 12, 7),
-    reason=(
-        "argparse stopped quoting choices in 3.13 (CPython gh-129019), "
-        "backported to 3.12.7. The reference snapshot reflects the "
-        "no-quote format, so older 3.12.x patches (3.12.0-3.12.6) print "
-        "quoted choices and fail. See commitizen-tools/commitizen#1864."
-    ),
+    ["--invalid-arg", "invalidCommand"],
 )
 @pytest.mark.usefixtures("python_version", "consistent_terminal_output")
 def test_invalid_command(util: UtilFixture, capsys, file_regression, arg):
@@ -59,6 +61,8 @@ def test_invalid_command(util: UtilFixture, capsys, file_regression, arg):
         util.run_cli(arg)
     out, err = capsys.readouterr()
     assert out == ""
+    if arg == "invalidCommand":
+        err = normalize_argparse_choice_quotes(err)
     file_regression.check(err, extension=".txt")
 
 

--- a/tests/test_cli/test_invalid_command_py_3_10_invalidCommand_.txt
+++ b/tests/test_cli/test_invalid_command_py_3_10_invalidCommand_.txt
@@ -1,4 +1,4 @@
 usage: cz [-h] [--config CONFIG] [--debug] [-n NAME] [-nr NO_RAISE]
           {init,commit,c,ls,example,info,schema,bump,changelog,ch,check,version}
           ...
-cz: error: argument {init,commit,c,ls,example,info,schema,bump,changelog,ch,check,version}: invalid choice: 'invalidCommand' (choose from 'init', 'commit', 'c', 'ls', 'example', 'info', 'schema', 'bump', 'changelog', 'ch', 'check', 'version')
+cz: error: argument {init,commit,c,ls,example,info,schema,bump,changelog,ch,check,version}: invalid choice: 'invalidCommand' (choose from init, commit, c, ls, example, info, schema, bump, changelog, ch, check, version)

--- a/tests/test_cli/test_invalid_command_py_3_11_invalidCommand_.txt
+++ b/tests/test_cli/test_invalid_command_py_3_11_invalidCommand_.txt
@@ -1,4 +1,4 @@
 usage: cz [-h] [--config CONFIG] [--debug] [-n NAME] [-nr NO_RAISE]
           {init,commit,c,ls,example,info,schema,bump,changelog,ch,check,version}
           ...
-cz: error: argument {init,commit,c,ls,example,info,schema,bump,changelog,ch,check,version}: invalid choice: 'invalidCommand' (choose from 'init', 'commit', 'c', 'ls', 'example', 'info', 'schema', 'bump', 'changelog', 'ch', 'check', 'version')
+cz: error: argument {init,commit,c,ls,example,info,schema,bump,changelog,ch,check,version}: invalid choice: 'invalidCommand' (choose from init, commit, c, ls, example, info, schema, bump, changelog, ch, check, version)

